pkg/network/ovn_kubernetes.go

```diff
@@ -561,12 +585,12 @@ func shouldRenderIPsec(conf *operv1.OVNKubernetesConfig, bootstrapResult *bootst
 	// MachineConfig IPsec extensions are needed for the ipsec deployment except
 	// when the containerized deployment is used in hypershift hosted clusters.
 	// We will rollout unless the user has rolled out its own.
-	renderIPsecMachineConfig = renderIPsecDaemonSet && !isUserIPsecMachineConfigPresent && !isHypershiftHostedCluster
+	renderIPsecMachineConfig = (isIPsecEnabled || renderIPsecDaemonSet) && !isUserIPsecMachineConfigPresent && !isHypershiftHostedCluster
```

Comment:
I think the comment above (or some comment) needs to be updated to reflect the fact that we now now somehow manage NS IPsec because before we only managed in in combination with EW IPsec.

Reply:
Like the comment above reads `MachineConfig IPsec extensions are needed for the ipsec deployment` but now is not only that since now we can enable NS IPsec with no ipsec deployment at all ....